frontend/src/components/common/form/LimitedSelect.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { memo, useEffect } from 'react';
```

Comment:
There wasn't too many changes to `FastSelect`... I am wondering if it should stay a separate component or add more to `FastSelect`. Not sure what the best practice is!

Reply:
If you're essentially displaying the current value and only allowing them to select specific values, I'd leave it as a single component.  This is a common scenario for dropdown/selects.